src/plugins/func/azureFunc.test.ts

```diff
@@ -0,0 +1,117 @@
+    const plugin = new AzureFuncPlugin(sls, options);
+    await invokeHook(plugin, "func:func");
+    expect(sls.cli.log).toBeCalledWith("Use the func plugin to add or remove functions within Function App");
+  })
+
+  describe("Add command", () => {
+
+    beforeAll(() => {
+      mockFs({
+        "myExistingFunction": {
+          "index.js": "contents",
+          "function.json": "contents",
+        },
+        "serverless.yml": MockFactory.createTestServerlessYml(true),
+      }, {createCwd: true, createTmp: true})
+    });
+  
+    afterAll(() => {
+      mockFs.restore();
+    });
+
+    it("returns with missing name", async () => {
+      const sls = MockFactory.createTestServerless();
+      const options = MockFactory.createTestServerlessOptions();
+      const plugin = new AzureFuncPlugin(sls, options);
+      await invokeHook(plugin, "func:add:add");
+      expect(sls.cli.log).toBeCalledWith("Need to provide a name of function to add")
+    });
+  
+    it("returns with pre-existing function", async () => {
+      const sls = MockFactory.createTestServerless();
+      const options = MockFactory.createTestServerlessOptions();
+      options["name"] = "myExistingFunction";
+      const plugin = new AzureFuncPlugin(sls, options);
+      await invokeHook(plugin, "func:add:add");
+      expect(sls.cli.log).toBeCalledWith(`Function myExistingFunction already exists`);
+    });
+  
+    it("creates function directory and updates serverless.yml", async () => {
+      const sls = MockFactory.createTestServerless();
+      const options = MockFactory.createTestServerlessOptions();
+      const functionName = "myFunction";
+      options["name"] = functionName;
+      const plugin = new AzureFuncPlugin(sls, options);
+      const mkdirSpy = jest.spyOn(fs, "mkdirSync");
+      await invokeHook(plugin, "func:add:add");
+      expect(mkdirSpy).toBeCalledWith(functionName);
+      const calls = (sls.utils.writeFileSync as any).mock.calls;
+      expect(calls[0][0]).toBe(path.join(functionName, "index.js"));
+      expect(calls[1][0]).toBe(path.join(functionName, "function.json"));
+      const expectedFunctionsYml = MockFactory.createTestFunctionsMetadata();
+      expectedFunctionsYml[functionName] = MockFactory.createTestFunctionMetadata();
+      expect(calls[2][0]).toBe("serverless.yml");
+      expect(calls[2][1]).toBe(MockFactory.createTestServerlessYml(true, expectedFunctionsYml));
+    });
+  });
+
+  describe("Remove command", () => {  
+    
+    beforeAll(() => {
+      mockFs({
+        "function1": {
+          "index.js": "contents",
+          "function.json": "contents",
+        },
+      }, {createCwd: true, createTmp: true});    
+    });
+  
+    afterAll(() => {
+      mockFs.restore();
+    });
+  
+    it("returns with missing name", async () => {
+      const sls = MockFactory.createTestServerless();
+      const options = MockFactory.createTestServerlessOptions();
```

Comment:
Q: Would it make sense to add a default name to the options and then remove/override as needed? Just thinking if we'll need it for other function tests if we add more functionality, like the option to include the extension bundles

Reply:
For sure. Let's add that in another PR